libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/CustomServerUrlEditor.java

```diff
@@ -93,7 +95,7 @@ public void onClick(View v) {
 					return;
 				}
 				final String val = validateInput(R.id.sf__picker_custom_url);
-				if (val == null) {
+				if (val == null || HttpUrl.parse(val) == null) {
```

Comment:
Shouldn't that move into validateInput?

Reply:
That's also shared with the label string (name).

Reply:
Yup - should move inside the following section in `validateInput`:
```
		if (editId == R.id.sf__picker_custom_url) {
			isInvalidValue = !URLUtil.isHttpsUrl(etVal.toString());
			if (isInvalidValue) {
				Toast.makeText(context, getString(R.string.sf__invalid_server_url),
						Toast.LENGTH_SHORT).show();
			}
		}
```

Reply:
We are doing url validation [here](https://github.com/Songroid/SalesforceMobileSDK-Android/blob/7efe3f31932aa565f52160538ced2de22d34e006/libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/CustomServerUrlEditor.java#L176) - maybe we should do all url validation there.